lib/subset.cc

```diff
@@ -236,12 +236,11 @@ void SubsetPartition::find_all_tags(
     unsigned int total = 0;
     unsigned int nfound = 0;
 
-    Traverser traverser(_ht);
     KmerSet keeper;		// keep track of traversed kmers
-
-    auto filter = [&] (Kmer& n) -> bool {
-        return !set_contains(keeper, n);
+    KmerFilter filter = [&] (const Kmer& n) -> bool {
+        return set_contains(keeper, n);
```

Comment:
Yeah -- turns out the behavior between the previous `Traverser` class and the new `AssemblerTraverser` with respect to `KmerFilter`s was inverted. For `Traverser`, a filter returned `true` to pass a filter, and for `AssemblerTraverser`, to fail. This was obviously somewhat confusing, and I decided that a k-mer _filter_ should universally return `true` if it aims to, well filter, a k-mer.